app/assets/javascripts/miq_flash.js

```diff
@@ -46,12 +61,48 @@ function add_flash(msg, level, options) {
   }
 
   $('#flash_msg_div').append(textDiv).show();
+
+  // remove dangling 'Show More' link when the alert msg is short.
+  if ( options.long_alert && ! checkElipsis(alertDiv) ) {
+    detailsLink.hide();
+  }
 }
 
 function clearFlash() {
   $('#flash_msg_div').empty();
 }
 
+function checkElipsis(element) {
+  var found = false;
+  var $c = element
+    .clone()
+    .css({display: 'inline', width: 'auto', visibility: 'hidden'})
+    .appendTo('body');
+  if ( $c.width() > element.width() ) {
+    found = true;
+  }
+
+  $c.remove();
+
+  return found;
+}
+
+function expandAlert(params) {
+  var viewMoreTxt = _("View More");
```

Comment:
In js, you need `__`, not `_` for i18n ;).

(`_` seems to work because it's lodash("string") -> ["string"].toString() -> "string" :))